### PR TITLE
feat/go

### LINKED ISSUE #51: feat: rewrite in Go
@azolus is currently working on a rewrite of this script in Go :tada:

### PR DESCRIPTION
**Pull request by azolus**
started rewriting dargstack in Go 🥳

requires go 1.18

this should eliminate some of the bash-related problems with mac and windows, once it's done

todo: proper documentation, dargstack build and deploy commands, and leveraging the flexibility of the go-implementation to clean up some of the bash related idiosyncracies in the dargstack-architecture

Resolves #51

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,15 +7,16 @@ on:
...
-      with:
-        scandir: './src'
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.17
+      - uses: actions/checkout@v3
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
   semantic-release:
     needs: shellcheck
```
Comment (dargmuesli) [inline .github/workflows/ci.yml:21]:
```suggestion
    needs: golangci
```

**Review by dargmuesli — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (azolus) [reply .github/workflows/ci.yml:21]: 👍🏼

**Comment by azolus** [comment]
I am putting this on hold, since I currently don't have time to finish this pr. Will get back to it next year.

**Comment by dargmuesli** [comment]
Closing due to inactivity

**Pull request closed without merging**